sympy/assumptions/handlers/order.py

```diff
@@ -56,43 +56,95 @@ def Basic(expr, assumptions):
 
     @staticmethod
     def Add(expr, assumptions):
-        """
-        Positive + Positive -> Positive,
-        Negative + Negative -> Negative
-        """
         if expr.is_number:
             return AskNegativeHandler._number(expr, assumptions)
 
         r = ask(Q.real(expr), assumptions)
         if r is not True:
             return r
 
-        nonpos = 0
-        for arg in expr.args:
-            if ask(Q.negative(arg), assumptions) is not True:
-                if ask(Q.positive(arg), assumptions) is False:
-                    nonpos += 1
+        args = [arg for arg in expr.args if not ask(Q.zero(arg), assumptions)]
+        if not args:
+            return False
+
+        inf = neg_inf = pos_inf = neg = npos = nneg = False
+
+        for arg in args:
+            n  = ask(Q.negative(arg), assumptions)
+            np = ask(Q.nonpositive(arg), assumptions)
+
+            infinite = ask(Q.infinite(arg), assumptions)
```

Comment:
Would it not suffice to know the sign of `arg`, whether finite or infinite?

Reply:
if any one argument is `+oo` it doesn't matter the other args are what sign. Similarly for `-oo` but if both `oo` and `-oo` happens to be present the result becomes ambiguous. Most of such ambiguities are handled.

Reply:
What if one argument is positive?

Reply:
I don't understand. All the arguments are checked. If positive and +oo are there then +oo would be resulted hence positive.

Reply:
And with -oo and a positive argument the result would be -oo always.

Reply:
We need to check what is overall effect of all the arguments is on the expression.

Reply:
Would it not be possible to use AskNonNegativeHandler to take care of positive and +oo?

Reply:
Yeah actually I can use the code is pretty much repetitive. Thanks for suggestion.

Reply:
I will see this case but I intended to make AskNegativeHandler and AskPositiveHandler the base handlers from them I would derive AskNonNegativeHandler and others. In this way much repetitive code can be reduced.